`shouldNotEmitPublisherPresenceFalseIfPublisherIsPresentFromTheStart` (`PublisherAndSubscriberTests`) failing with "first publisherPresence value should be true"

I've only seen one failure so far for this test, but am creating an issue so we can track it.

Failing from our Android emulation workflow. This link will only survive as long as GitHub keeps the test run logs, so I'll also quote from the output...

[At API Level 24](https://github.com/ably/ably-asset-tracking-android/actions/runs/3685077251/jobs/6235588965#step:4:960) (https://github.com/ably/ably-asset-tracking-android/pull/842):

```
com.ably.tracking.tests.PublisherAndSubscriberTests > shouldNotEmitPublisherPresenceFalseIfPublisherIsPresentFromTheStart[test(AVD) - 7.0] FAILED 
	java.lang.AssertionError: first publisherPresence value should be true
	at org.junit.Assert.fail(Assert.java:89)
```

[At API Level 21](https://github.com/ably/ably-asset-tracking-android/actions/runs/3758334740/jobs/6386549727#step:4:858) (https://github.com/ably/ably-asset-tracking-android/pull/856):

```
com.ably.tracking.tests.PublisherAndSubscriberTests > shouldNotEmitPublisherPresenceFalseIfPublisherIsPresentFromTheStart[test(AVD) - 5.0.2] FAILED 
	java.lang.AssertionError: first publisherPresence value should be true
	at org.junit.Assert.fail(Assert.java:89)
```

[At API Level 21](https://github.com/ably/ably-asset-tracking-android/actions/runs/3846090573/jobs/6550991350#step:4:732):

```
com.ably.tracking.tests.PublisherAndSubscriberTests > shouldNotEmitPublisherPresenceFalseIfPublisherIsPresentFromTheStart[test(AVD) - 5.0.2] FAILED 
	java.lang.AssertionError: first publisherPresence value should be true
	at org.junit.Assert.fail(Assert.java:89)
Tests on test(AVD) - 5.0.2 failed: There was 1 failure(s).

> Task :integration-testing-app:connectedDebugAndroidTest FAILED
```

While working locally on #259 I had this test fail when I ran this group of tests.

I used `./gradlew -Pandroid.testInstrumentationRunnerArguments.class=com.ably.tracking.tests.PublisherAndSubscriberTests :integration-testing-app:connectedDebugAndroidTest` and got:

```
com.ably.tracking.tests.PublisherAndSubscriberTests > shouldNotEmitPublisherPresenceFalseIfPublisherIsPresentFromTheStart[Nexus_S_API_22(AVD) - 5.1.1] FAILED
        java.lang.AssertionError: first publisherPresence value should be true
        at org.junit.Assert.fail(Assert.java:89)
Tests on Nexus_S_API_22(AVD) - 5.1.1 failed: There was 1 failure(s).

> Task :integration-testing-app:connectedDebugAndroidTest FAILED
```

Where a bit more detail was present in the report:

```
java.lang.AssertionError: first publisherPresence value should be true
at org.junit.Assert.fail(Assert.java:89)
at org.junit.Assert.assertTrue(Assert.java:42)
at com.ably.tracking.tests.PublisherAndSubscriberTests.shouldNotEmitPublisherPresenceFalseIfPublisherIsPresentFromTheStart(PublisherAndSubscriberTests.kt:294)
```

at (`git rev-parse HEAD`): 3b15763d5220ff733571e678b4cc69d341a1fed5

--- a/integration-testing-app/src/androidTest/java/com/ably/tracking/tests/PublisherAndSubscriberTests.kt
+++ b/integration-testing-app/src/androidTest/java/com/ably/tracking/tests/PublisherAndSubscriberTests.kt
@@ -251,6 +251,7 @@ class PublisherAndSubscriberTests {
     @Test
     fun shouldNotEmitPublisherPresenceFalseIfPublisherIsPresentFromTheStart() {
         // given
+        val publisherViewsTrackableAsOnline = UnitExpectation("publisher views trackable as online")
         val subscriberEmittedPublisherPresentExpectation = UnitExpectation("subscriber emitted publisher present")
         val context = InstrumentationRegistry.getInstrumentation().targetContext
         val trackableId = UUID.randomUUID().toString()
@@ -261,8 +262,13 @@ class PublisherAndSubscriberTests {
         // create publisher and start tracking
         val publisher = createAndStartPublisher(context, sendResolution = true)
         runBlocking {
-            publisher.track(Trackable(trackableId))
+            publisher.track(Trackable(trackableId)).onEach { trackableState ->
+                if (trackableState == TrackableState.Online) {
+                    publisherViewsTrackableAsOnline.fulfill()
+                }
+            }.launchIn(scope)
         }
+        publisherViewsTrackableAsOnline.await()
 
         // create subscriber and listen for publisher presence
         var subscriber: Subscriber
@@ -290,6 +296,7 @@ class PublisherAndSubscriberTests {
         }
 
         // then
+        publisherViewsTrackableAsOnline.assertFulfilled()
         subscriberEmittedPublisherPresentExpectation.assertFulfilled()
         Assert.assertTrue("first publisherPresence value should be true", publisherPresentValues.first())
     }

